deployments/deploy-tests.ps1

```diff
@@ -1,99 +1,122 @@
-    $kvOptions.Name="KV-MYAEA-KKZH-ADM-C1-D01" 
-    $kv = New-KeyVaultDefinition -Options $kvOptions
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "ADM" -EnvironmentLetter "D" -Region "CentralUs"
 
-    $laOptions = New-LogAnalyticsOptions
-    $laOptions.Name="LA-MYAEA-KKZH-ADM-C1-D01" 
-    $la = New-LogAnalyticsDefinition -Options $laOptions
+    #name, az context built from deployment context params above
+    $kv = New-PfKeyVaultBuild
+    $kv.Options.Name = "testing"
+    Set-PfAzureContext -CompanyAbbreviation "MYAEA" -GroupAbbreviation "KKZH" -TeamAbbreviation "DM" -EnvironmentLetter "D" -Region "CentralUs"
 
-    $deployContext.ResourceDefinitions.AddRange(@($kv,$la))
-    #$deployContext.Deploy()
+    #name, az context built from deployment context params above
+    $la = New-PfLogAnalyticsBuild
+    $la.Options.ResourceGroupName="NewRg"
+    Deploy-PfDeploymentContext
 
     ##############################################################################################################
     #
-    #               Deploy a single resource with context and deployment context
+    #               Deploy multiple resources with context and deployment context
     #
     #############################################################################################################
 
     #Still need to write the deploy method, but the thought here is that you get the sub and rg from the
     #current context so all you need to do is define the name and other options
 
-    $deployContext2 = New-PfDeploymentContext
-
-    Set-AzCloudContext -SubscriptionName "MYAEA-KKZH-D" -ResourceGroupName "RG-MYAEA-KKZH-ADM-D"
+    New-PfDeploymentContext
 
-    $kvOptions = New-KeyVaultOptions
-    $kvOptions.Name="KV-MYAEA-KKZH-ADM-C1-D01" 
-    $kv = New-KeyVaultDefinition -Options $kvOptions
+    Set-PfAzureContext -SubscriptionName "MYAEA-KKZH-D" -ResourceGroupName "RG-MYAEA-KKZH-ADM-D" -Region "CentralUs"
 
-    $deployContext2.ResourceDefinitions.Add($kv)
-    #$deployContext2.Deploy()
+    $kv = New-PfKeyVaultBuild
+    $kv.Options.Name="KV-MYAEA-KKZH-ADM-C1-D01" 
 
+    $la = New-PfLogAnalyticsBuild
+    $la.Options.Name="LA-MYAEA-KKZH-ADM-C1-D01" 
 
+    Deploy-PfDeploymentContext
 
     ##############################################################################################################
     #
-    #               Deploy a single resource with context
+    #               Deploy a single resource with context and deployment context
     #
     #############################################################################################################
 
     #Still need to write the deploy method, but the thought here is that you get the sub and rg from the
     #current context so all you need to do is define the name and other options
 
-    Set-AzCloudContext -SubscriptionName "MYAEA-KKZH-D" -ResourceGroupName "RG-MYAEA-KKZH-ADM-D"
+    New-PfDeploymentContext
```

Comment:
ok just finally realized implications of the split deploy and azure contexts walking through deploy scripts.  would it make any sense to move all the 'custom' fields into deploy context?

    [string]$TeamAbbreviation
    [string]$CompanyAbbreviation
    [string]$GroupAbbreviation
    [string]$EnvironmentLetter

Reply:
custom or 'pf' fields really i guess

Reply:
So DeployContext is just that.  It is just like a "using" statement saying any resources after this will be added to the context and potentially deployed.  I don't want to limit a deployment to just a single "Group" or "Environment" or even "company".  I would like that to be driven on the resources.  So now the deployment context just iterates through the resources and know this resource goes to production.  The resource is development.  This resource is for the dev group, and this resource is for the db group.